frontend/actions/now_playing_actions.js

```diff
@@ -1,19 +1,40 @@
-export const TOGGLE_PLAY = 'TOGGLE_PLAY'
-export const PLAY_ALBUM = 'PLAY_ALBUM'
-export const PLAY_PLAYLIST = 'PLAY_PLAYLIST'
+export const TOGGLE_PLAY = "TOGGLE_PLAY";
+export const PUSH_PLAY = "PUSH_PLAY";
+export const QUEUE_ARTIST = "QUEUE_ARTIST";
+export const PLAY_ARTIST = "PLAY_ARTIST";
 
 const togglePlay = () => ({
-    type: TOGGLE_PLAY,
-})
+	type: TOGGLE_PLAY,
+});
 
-const playAlbum = () => ({
-    type: PLAY_ALBUM,
-})
+const pushPlay = () => ({
+	type: PUSH_PLAY,
+});
 
-const playPlaylist = () => ({
-    type: PLAY_PLAYLIST,
-})
+const queueArtist = (queueObj) => ({
+	//{allSongs:arr, sourceType:str, extractedUrlParams:numStr}
+	type: QUEUE_ARTIST,
+	allSongs: queueObj.allSongs,
+	sourceType: queueObj.sourceType,
+	extractedUrlParams: queueObj.extractedUrlParams,
```

Comment:
Since the Player component doesn't have a React-Router path for itself, it currently doesn't have access to the url/path/params of the current view. Creating this queueObj in ArtistShow and passing it into Redux would allow the source of the songs added to be accessible across the app.

Reply:
Oh I did put it but in a different spot lol: https://github.com/imartinez921/versify_full-stack/pull/97/files#:~:text=%7D%3B%20//%20provides,Implement%20queue%20view